cpp/examples/jsonschema/writer.cpp

```diff
@@ -0,0 +1,98 @@
+// Example of writing JSON messages to an MCAP file in C++.
+// Writes a topic of Point2 messages.
+#define MCAP_IMPLEMENTATION
+#include "mcap/writer.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <chrono>
+#include <cmath>
+#include <iostream>
+
+#define NUM_FRAMES 100
+#define NS_PER_MS 1000000
+
+static const char* SCHEMA_NAME = "foxglove.Point2";
+static const char* SCHEMA_TEXT =
+  "{ \
```

Comment:
use `R"(...)"` [raw string literal](https://en.cppreference.com/w/cpp/language/string_literal) for readability (to avoid needing escapes)